File: aat-awt/src/main/java/ch/bailu/aat_awt/logger/SL4JCache.java
package ch.bailu.aat_awt.logger;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.HashMap;
import java.util.Map;

public class SL4JCache {
    private final Map<String, Logger> loggers = new HashMap<>();

    public static final SL4JCache SELF = new SL4JCache();

    private SL4JCache() {}


    public Logger get(Class c) {
        return get(c.getName());
    }

    public Logger get(String name) {
        Logger result;

        if (loggers.containsKey(name)) {
            result =  loggers.get(name);
        } else {
            result = LoggerFactory.getLogger(name);
            loggers.put(name, result);
        }

        return result;
    }
}

File: aat-gtk/src/main/kotlin/ch/bailu/aat_gtk/logger/SL4JCache.kt
package ch.bailu.aat_gtk.logger

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import java.util.HashMap


object SL4JCache {
    private val loggers: MutableMap<String, Logger> = HashMap()

    operator fun get(c: Class<*>): Logger {
        return get(c.name)
    }

    operator fun get(name: String): Logger {
        var result = loggers[name]

        return if (result == null) {
            result = LoggerFactory.getLogger(name)
            loggers[name] = result
            result
        } else {
            result
        }
    }
}
